ci-operator/step-registry/ipi/deprovision/azure/resourcegroup/ipi-deprovision-azure-resourcegroup-commands.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+existing_rg=$(/tmp/yq r "${CONFIG}" 'platform.azure.resourceGroupName')
+
+# az should already be there
+az version
+
+# set the parameters we'll need as env vars
+AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+AZURE_AUTH_CLIENT_ID="$(cat ${AZURE_AUTH_LOCATION} | jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(cat ${AZURE_AUTH_LOCATION} | jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(cat ${AZURE_AUTH_LOCATION} | jq -r .tenantId)"
+
+# log in with az
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}"
+
+# delete resource group in case it still exists
+echo "Deleting resource group: ${existing_rg}"
+az group delete -y --no-wait -n "${existing_rg}" &>/dev/null
```

Comment:
I think we **do** want to wait.
Also, we don't want to discard the output. We want to be able to see the output in case there are problems. I am assuming that the intention here is to not fail when the resource group does not exist.
```suggestion
if [ "$(az group exists -n "${existing_rg}")" == "true" ]
then
  az group delete -y -n "${existing_rg}"
fi
```